### ISSUE TITLE
Make Storage Factory mockable

### ISSUE THREAD

**Issue by emulgeator** [szeber/yapep_base#20]
Storage Factory is a simple static class which makes it impossible to mock.

**Comment by szeber** [comment]
I agree, probably best to move it in the DI container, and make the methods non static.

**Comment by emulgeator** [comment]
Before even doing it, please give an example about the DI solution you mentioned before (when we spoke)
